packages/core/src/machines/authMachine.ts

```diff
@@ -0,0 +1,172 @@
+import { assign, createMachine, forwardTo, spawn } from 'xstate';
+import { Auth, Amplify } from 'aws-amplify';
+import { AuthContext, AuthEvent } from '../types';
+import { inspect } from '@xstate/inspect';
+import {
+  signInActor,
+  signUpActor,
+  signOutActor,
+  resetPasswordActor,
+} from './actors';
+import { stopActor } from './actions';
+
+// TODO: Remove this before it's merged.
+if (typeof window !== 'undefined') {
+  inspect({
+    // options
+    // url: 'https://statecharts.io/inspect', // (default)
+    iframe: false, // open in new window
+  });
+}
+
+export const authMachine = createMachine<AuthContext, AuthEvent>(
+  {
+    id: 'auth',
+    initial: 'idle',
+    context: {},
+    states: {
+      // See: https://xstate.js.org/docs/guides/communication.html#invoking-promises
+      idle: {
+        invoke: [
+          {
+            // TODO Wait for Auth to be configured
+            src: 'getCurrentUser',
+            onDone: {
+              actions: 'setUser',
+              target: 'authenticated',
+            },
+            onError: 'signIn',
+          },
+          {
+            src: 'getAmplifyConfig',
+            onDone: {
+              actions: 'setAuthConfig',
+            },
+          },
+        ],
+      },
+      signIn: {
+        entry: 'spawnSignInActor',
+        exit: stopActor('signInActor'),
+        on: {
+          SIGN_UP: 'signUp',
+          RESET_PASSWORD: 'resetPassword',
+          'done.invoke.signInActor': [
```

Comment:
This is the xstate built-in emitted event when signInActor completes. Need to use this because actors are spawned and not invoked, thus doesn't have an explicit `onDone` handler.